### PR TITLE
CASM-4727: Migrate all products to latest version of cpc

### PR DESCRIPTION
**Pull request by Pankhuri-Rajesh**
## Summary and Scope

The changes are related to the new format of cray-product-catalog . The PR involves changes for three JIRAs attached below.

Two of them are bug fixes (4741 and 4743).

The third one involves design change to make the format of cray-product-catalog after migration more efficient and compatible. (4731) 
More details are present in the JIRA regarding all the code changes.

Also there is a change in setup.py to make use of the cpc module in iuf-cli code .

## Issues and Related PRs

* Resolves [CASM-4731](https://jira-pro.it.hpe.com:8443/browse/CASM-4731), [CASM-4743](https://jira-pro.it.hpe.com:8443/browse/CASM-4743), [CASM-4741](https://jira-pro.it.hpe.com:8443/browse/CASM-4741)

### Tested on:

  * starlord

## Pull Request Checklist

- [x] Target branch correct
- [x] Testing is appropriate and complete, if applicable

### CODE REVIEW

**Review by rajeshranganathan-hpe — APPROVED** [review]

File: cray_product_catalog/catalog_delete.py [hunk 1]
```diff
@@ -219,7 +219,29 @@ def modify(self):
             return
 
         self.__modify_main_cm()
-        self.__modify_product_cm()
+
+        """Invoke __modify_product_cm function only if the product_cm is present
+
+        Before attempting to delete a configmap, first verify if for the product there is a product_cm present or not.
+        If not, give a warning message and continue
+        """
```
Comment (mharding-hpe) [inline cray_product_catalog/catalog_delete.py:227]: Triple quotes should not be used for regular code comments (function/method/class/module docstrings are an exception).
```suggestion
        # Invoke __modify_product_cm function only if the product_cm is present

        # Before attempting to delete a ConfigMap, first verify if for the product there is a product_cm present or not.
        # If not, give a warning message and continue
```

File: cray_product_catalog/migration/main.py [hunk 2]
```diff
@@ -44,8 +44,31 @@
 LOGGER = logging.getLogger(__name__)
 
 
+# function to check if configmap is already migrated
+def is_migrated():
```
Comment (mharding-hpe) [inline cray_product_catalog/migration/main.py:48]: This should be a function docstring, not a comment above the function
```suggestion
def is_migrated():
   """
   Check if ConfigMap is already migrated.
   Returns True if so, False if not.
   """
```

File: cray_product_catalog/migration/main.py [hunk 3]
```diff
@@ -44,8 +44,31 @@
...
+    if labels and labels.get(PRODUCT_CATALOG_CONFIG_MAP_LABEL_KEY) == PRODUCT_CATALOG_CONFIG_MAP_NAME:
+        return True
+
+    return False
+
+
 def main():
     """Main function"""
+
+    # check if configmap has already been migrated
```
Comment (mharding-hpe) [inline cray_product_catalog/migration/main.py:67]:
```suggestion
    # Check if ConfigMap has already been migrated
```

**Review by mharding-hpe — CHANGES_REQUESTED** [review]

**Review by mharding-hpe — APPROVED** [review]

File: cray_product_catalog/migration/main.py [hunk 4]
```diff
@@ -44,8 +44,34 @@
 LOGGER = logging.getLogger(__name__)
 
 
+def is_migrated():
+   """
+   Check if ConfigMap is already migrated.
+   Returns True if so, False if not.
+   """
+    config_map_obj = ConfigMapDataHandler()
```
Comment (mharding-hpe) [inline cray_product_catalog/migration/main.py:52]:
```suggestion
def is_migrated():
    """
    Check if ConfigMap is already migrated.
    Returns True if so, False if not.
    """
    config_map_obj = ConfigMapDataHandler()
```

**Review by mharding-hpe — APPROVED** [review]

**Pull request merged**